p2p/src/net/mock/transport/traits/socket.rs

```diff
@@ -52,5 +53,5 @@ pub trait TransportSocket: Send + Sync + 'static {
     async fn bind(&self, address: Vec<Self::Address>) -> Result<Self::Listener>;
 
     /// Open a connection to the given address.
-    async fn connect(&self, address: Self::Address) -> Result<Self::Stream>;
+    fn connect(&self, address: Self::Address) -> BoxFuture<'static, crate::Result<Self::Stream>>;
```

Comment:
Nitpick: perhaps the comment should be updated to reflect the changes. Something like "...returns a future that opens a connection...".

Reply:
Sure, will do, thanks. Will do in next PR (don't want to wait for CI checks).